nav2_costmap_2d/plugins/obstacle_layer.cpp

```diff
@@ -243,6 +243,33 @@ void ObstacleLayer::onInitialize()
       observation_notifiers_.back()->setTargetFrames(target_frames);
     }
   }
+  setParamCallbacks();
+}
+
+void
+ObstacleLayer::setParamCallbacks()
+{
+  if (param_subscriber_) {
```

Comment:
Where are the observation source parameters? Those are the really valuable ones to have reconfigurable

Reply:
This PR is mainly concerned with parity with the ROS1 dynamic parameters. I don't believe the observation source parameters were included there. So we can access and add them if we want, but that wasn't my starting point here.

Reply:
I’m fairly certain it was- or at minimum when one sensor was available definitely was. Right now there isn’t all the params with Cbs for that to be true

Reply:
I scrubbed the ROS1 navigation code and I've already included all the costmap plugin reconfigure parameters in this PR, so no observation sources.

Reply:
If these arent happening at the observation level, how do you see this having any affect on the output?

Reply:
I'm not sure I understand what you mean. These parameters in particular are not on the observation source.

Reply:
They can be set differently for each individual oberservation source. They are not global parameters to a layer except when an individual observation source didnt set it and in that case it defaults to a global value

Reply:
These parameters (`enabled_, footprint_clearing_enabled_, max_obstacle_height_, combination_method_`) in the obstacle layer do have an effect globally, they aren't simply defaults for the individual observation sources. Take a look at the functions `updateCosts`, `updateBounds` and `updateFootprint`. These were the set of parameters that were reconfigurable in the orginal navigation stack.

Reply:
@SteveMacenski Do you think adding support for the observation source parameters could be a part of a separate PR? I'm currently running these PRs as a "mostly" ROS1 parity check, but open to start the assessment & re-design early if we think that's more appropriate.

Reply:
I think if you just removed the obstacle / voxel layer stuff the rest is fine as is if that’s just entirely added as future work

Reply:
Eh. Whatever. File a ticket for observation params for obstacle/voxel and we can call it.